### PR TITLE
Incidents support

### PR DESCRIPTION
**Pull request by imxeno**

### CODE REVIEW

File: packages/config/src/projects/layer2s/aevo.ts [hunk 1]
```diff
@@ -72,6 +72,8 @@ export const aevo: Layer2 = opStackL2({
       link: 'https://twitter.com/aevoxyz/status/1750013642278633510',
       date: '2024-01-16T00:00:00.00Z',
       description: 'Aevo starts using Celestia for data availability.',
+      // TODO: for testing purposes, remove this type later
```
Comment (torztomasz) [inline packages/config/src/projects/layer2s/aevo.ts:75]: let's remove then

File: packages/frontend/src/pages/project/bridge/props/getProjectDetails.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ export function getProjectDetails(
       props: {
         milestones: bridge.milestones,
         id: 'milestones',
```
Comment (torztomasz) [inline packages/frontend/src/pages/project/bridge/props/getProjectDetails.ts:71]: let's change the id maybe?

File: packages/frontend/src/pages/project/components/sections/MilestonesSection/MilestonesSection.tsx [hunk 3]
```diff
@@ -26,36 +28,72 @@ export function MilestonesSection(props: MilestonesSectionProps) {
...
-                  )}
-                  <Link className="text-sm" href={milestone.link} showArrow>
-                    Learn more
-                  </Link>
+            .map((milestone, i) => {
+              // Milestones may have a type of 'general' or 'incident'
+              // When the type is undefined, it is considered 'general'
+
+              const milestoneType = milestone.type || 'general'
+              const oldMilestone = props.milestones.at(i + 1)
```
Comment (torztomasz) [inline packages/frontend/src/pages/project/components/sections/MilestonesSection/MilestonesSection.tsx:41]: `previousMilestone` would be better

File: packages/config/src/common/Milestone.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
+export type MilestoneType = 'general' | 'incident'
+
 export interface Milestone {
   name: string
   link: string
   date: string
   description?: string
+  type?: MilestoneType
```
Comment (torztomasz) [inline packages/config/src/common/Milestone.ts:8]: hmm, I am just thinking if it should be optional, probably not

File: packages/frontend/src/scripts/charts/styles.ts [hunk 5]
```diff
@@ -73,6 +74,10 @@ export const POINT_CLASS_NAMES: PointClassNames = {
     type: 'div',
     className: 'size-2 rotate-45 border-2 border-green-500 bg-green-700',
   },
+  incident: {
+    type: 'div',
+    className: 'size-2 rotate-45 border-2 border-red-700 bg-red-800',
+  },
```
Comment (torztomasz) [inline packages/frontend/src/scripts/charts/styles.ts:80]: its not a triangle, is it ok?

File: packages/frontend2/src/app/_components/projects/sections/milestones-section.tsx [hunk 6]
```diff
@@ -52,40 +57,85 @@ export function MilestonesSection({
...
+              return null
+            }
+
+            // Milestones may have a type of 'general' or 'incident'
+            // When the type is undefined, it is considered 'general'
+
+            const milestoneType = milestone.type ?? 'general'
+            const oldMilestone = props.milestones.at(i + 1)
+            const oldMilestoneType =
+              oldMilestone && (oldMilestone.type ?? 'general')
```
Comment (torztomasz) [inline packages/frontend2/src/app/_components/projects/sections/milestones-section.tsx:86]: same, would name it previous

**Review by torztomasz — CHANGES_REQUESTED** [review]
make sure all the comments are resolved on frontend2 and frontend

File: packages/config/src/common/Milestone.ts (hunk 4 above)
Comment (imxeno) [reply packages/config/src/common/Milestone.ts:8]: This means updating all milestones in the entire codebase, are you sure? :)

File: packages/frontend/src/scripts/charts/styles.ts (hunk 5 above)
Comment (imxeno) [reply packages/frontend/src/scripts/charts/styles.ts:80]: Triangles look weird on the chart, that's why I left it like this.

**Review by torztomasz — APPROVED** [review]

**Pull request merged**